apps/web-giddh/src/app/services/companyService.service.ts

```diff
@@ -399,15 +399,99 @@ export class CompanyService {
+     * @returns
+     * @memberof CompanyService
+     */
+    public getIntegratedBankInCompany(companyUniqueName: string): Observable<BaseResponse<any, any>> {
+        return this._http.get(this.config.apiUrl + COMPANY_API.GET_COMPANY_INTEGRATED_BANK_LIST
+            .replace(':companyUniqueName', encodeURIComponent(companyUniqueName))).pipe(map((res) => {
+                let data: BaseResponse<any, string> = res;
+                return data;
+            }), catchError((e) => this.errorHandler.HandleCatch<string, any>(e, ReportsRequestModel)));
+    }
+
+    /**
+     * Get all integrated banks data from bank server
+     *
+     * @param {string} companyUniqueName company unique name
+     * @param {string} urn urn number
+     * @returns
+     * @memberof CompanyService
+     */
+    public getAllBankDetailsOfIntegrated(companyUniqueName: string, urn: string): Observable<BaseResponse<any, any>> {
+        let url = this.config.apiUrl + COMPANY_API.GET_COMPANY_INTEGRATED_BANK_DETAILS.replace(':companyUniqueName', encodeURIComponent(companyUniqueName)).replace(':urn', urn);
+        return this._http.get(url).pipe(map((res) => {
+            let data: BaseResponse<any, string> = res;
             return data;
-        }), catchError((e) => this.errorHandler.HandleCatch<string, ReportsRequestModel>(e, ReportsRequestModel)));
-}
+        }), catchError((e) => this.errorHandler.HandleCatch<string, any>(e, ReportsRequestModel)));
+    }
+
+    /**
+     * Bulk pay vendor API call
+     *
+     * @param {string} companyUniqueName  company unique name
+     * @param {GetOTPRequest} bankTransferRequest request object
+     * @returns {Observable<BaseResponse<string, GetOTPRequest>>}
+     * @memberof CompanyService
+     */
+    public bulkVendorPayment(companyUniqueName: string, bankTransferRequest: GetOTPRequest): Observable<BaseResponse<BulkPaymentResponse, GetOTPRequest>> {
+        if (this.companyUniqueName) {
+            return this._http.post(this.config.apiUrl + COMPANY_API.BULK_PAYMENT.replace(':companyUniqueName', encodeURIComponent(companyUniqueName)), bankTransferRequest).pipe(map((res) => {
+                return res;
+            }), catchError((e) => this.errorHandler.HandleCatch<BulkPaymentResponse, GetOTPRequest>(e, bankTransferRequest)));
+        } else {
+            return observableEmpty();
+        }
+    }
+
+    /**
+     * Bulk pay vendor API call
+     *
+     * @param {string} companyUniqueName  Company unique name
+     * @param {string} urn Selected bank urn number
+     * @param {BulkPaymentConfirmRequest} requestObject Request object
+     * @returns {Observable<BaseResponse<BulkPaymentResponse, BulkPaymentConfirmRequest>>}
+     * @memberof CompanyService
+     */
+    public bulkVendorPaymentConfirm(companyUniqueName: string, urn: string, requestObject: BulkPaymentConfirmRequest): Observable<BaseResponse<BulkPaymentResponse, BulkPaymentConfirmRequest>> {
+        if (this.companyUniqueName) {
```

Comment:
I think this is incorrect condition as method accepts a parameter for `companyUniqueName` and the same is used in API call so we should check that instead of service member `companyUniqueName`